### PR TITLE
Fix #524, Fix #534 

### LINKED ISSUE #524: Viewer fails to initialize inside cross-origin iframe
**Describe the bug**
The viewer can't be initialized when rendered inside a cross-origin iframe.
This is due to a window.top event listener and is easy to fix.

**To Reproduce**

1. Create a folder `miew-test`

1. Set up two web servers pointing to the same folder, to simulate a cross origin iframe.
   
    In one terminal run:

        python -m http.server 8000
    
    In another terminal run:

        python -m http.server 8001
    
    > Note: Python 3 assumed. For Mac use `python3`

1. Create `viewer.html` with a functioning viewer:

```
<!DOCTYPE html>
<html lang="en">
    <head>
        <link rel="stylesheet" href="https://unpkg.com/miew@0.9.0/dist/Miew.min.css" />
        <script src="https://unpkg.com/@babel/polyfill@7/dist/polyfill.min.js"></script>
        <script src="https://unpkg.com/lodash@4.17.15/lodash.js"></script>
        <script src="https://unpkg.com/three@0.112.1/build/three.min.js"></script>
        <script src="https://unpkg.com/miew@0.9.0/dist/Miew.min.js"></script>
    </head>
    <body>
        <h1>Miew Viewer</h1>
        <div class="miew-container" style="width: 640px; height: 480px"></div>

        <script>
            ;(function () {
                var viewer = new Miew({ load: '1CRN' })
                if (viewer.init()) {
                    viewer.run()
                }
            })()
        </script>
    </body>
</html>
```

1. Create `index.html` with an iframe pointing at the viewer:

```
<!DOCTYPE html>
<html lang="en">
    <head>
    </head>
    <body>
        <iframe src="http://localhost:8001/viewer.html" width="800" height="600"></iframe>
    </body>
</html>
```

1. Open http://localhost:8000

**Expected behavior**
The viewer should initialize successfully.

**Actual behavior**
The viewer doesn't initialize successfully.

**Source of the problem**
The two lines below are responsible. Because we're referring to window.top, this causes a CORS error blocking the further initialization of the viewer. Replacing `window.top` with `window` fixes the problem. If it's important to keep `window.top`, logic should be added to prevent the initialization from crapping out completely.

`Miew.js`

```
window.top.addEventListener('keydown', (event) => {
  self._onKeyDown(event);
});

window.top.addEventListener('keyup', (event) => {
  self._onKeyUp(event);
});

```
If `window.top` can be replaced, I presume it should also be updated in `ObjectControls.js`:

```
ObjectControls.prototype.getKeyBindObject = function () {
  return window.top;
};
```

**Screenshots**
Screenshot 2024-02-27 at 11 47 26 AM
Screenshot 2024-02-27 at 11 47 32 AM
Screenshot 2024-02-27 at 11 48 25 AM

**Context**
 - OS: MacOS Sonoma 14.3.1 (23D60)
 - Browser: Google Chrome 
 - Version: Version 118.0.5993.117 (Official Build) (arm64)
 - npm Miew version: 0.10.1

**Additional context**
If you're ok with replacing `window.top` with `window` I can submit a pull request.

### LINKED ISSUE #534: Miew will disabled s/+/- keys on pages where it's initialized
**Describe the bug**
The `s` key as well as the numeric keyboard's `+` / `-` keys are disabled when you try to type on a page where Miew is initialized.

**To Reproduce**
- Initiate Miew with default settings
- Try typing in an input field on the same page as the Miew viewer

**Expected behavior**
`s` key should type 's' character.

**Additional context**
Responsible code:
```
Miew.prototype._onKeyDown = function (event) {
    [...]
    switch (event.keyCode) {
    case 'S'.charCodeAt(0):
        event.preventDefault();
        event.stopPropagation();
        settings.set('ao', !settings.now.ao);
        this._needRender = true;
        break;
    [...]
}
```

In the change log there's a mention of when and why this was implemented:

```
## [0.7.4] - 2017-10-16
- Prevent 's' key from appearing in the terminal when it is closed.
```

### PR DESCRIPTION
**Pull request by themoenen**
## Description

This pull request addresses issues #524 and #534:

- Allow Miew to be initialized within a CORS iframe (essential for Jupyter Notebook support)
- Prevent certain keys from being blocked when typing in text fields on pages where Miew is initialized

Should be reviewed by someone more familiar with the codebase to make sure there's no unintended consequences.

## Type of changes

- Bug fix (non-breaking change that fixes an issue)

## Checklist

- [x] I have read [CONTRIBUTING](https://github.com/epam/miew/blob/master/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/epam/miew/blob/master/CODE_OF_CONDUCT.md) guides.
- [x] I have followed the code style of this project.
- [x] I have run `yarn run ci`: lint and tests pass locally with my changes.
- [ ] I have added tests that prove my fix/feature works _OR_ The changes do not require updated tests.
- [x] I have added the necessary documentation _OR_ The changes do not require updated docs.

### CODE REVIEW

File: packages/lib/dist/Miew.module.js
Comment (paulsmirnov) [inline packages/lib/dist/Miew.module.js:1]: The `dist/` folder is updated only on new version releases (and corresponds to the version published in NPM) so I don't think we should edit files here manually in PRs.

File: packages/lib/src/Miew.js [hunk 1]
```diff
@@ -2888,24 +2897,30 @@ Miew.prototype._onKeyDown = function (event) {
       }
       break;
     case 'S'.charCodeAt(0):
-      event.preventDefault();
-      event.stopPropagation();
+      if (!['TEXTAREA', 'INPUT'].includes(document.activeElement?.tagName)) {
```
Comment (paulsmirnov) [inline packages/lib/src/Miew.js:2900]: I wonder if there are other tag names to support. Or even there could be an arbitrary DIV element (a React component) waiting for the S key and this solution wouldn't help. This fix might be a short-term solution if `miew.enableHotKeys(false)` doesn't work for you.

The long-term solution I see is:
- move all key handlers to the demo apps providing proper public methods in the library;
- disable the `S` key completely as it looks more like a debugging key;

However, it is just my quick glance with a dump of current thoughts.

File: packages/lib/.DS_Store
Comment (paulsmirnov) [inline packages/lib/.DS_Store:1]: It seems a random file has arrived from a MacOS. We'd better put the name into `.gitignore`.

File: package.json [hunk 2]
```diff
@@ -23,6 +23,6 @@
   },
   "packageManager": "yarn@3.6.3",
   "resolutions": {
-    "nomnom": "^1.8.1" 
+    "nomnom": "^1.8.1"
```
Comment (paulsmirnov) [inline package.json:26]: An irrelevant change but a good catch nevertheless.

**Review by paulsmirnov — COMMENTED** [review]
We appreciate your help on this! I am sorry I am not ready to give it a full thought currently but here are my quick notes on the changes.

**Comment by paulsmirnov** [comment]
Thank you for your contribution. Unfortunately, the necessary revisions have not been made, and we have decided to close this PR for now. We appreciate your effort and might revisit implementing similar changes in the future. Thank you for your understanding!

**Comment by themoenen** [comment]
My apologies. I think #550 solves this in a more elegant way anyway. Deleting this fork.

**Pull request closed without merging**